text_extensions_for_pandas/array/token_table.py

```diff
@@ -0,0 +1,84 @@
+#
+#  Copyright (c) 2021 IBM Corp.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+#
+# token_table.py
+#
+# Part of text_extensions_for_pandas
+#
+# Data structures for managing collections of tokens. Used internally
+# by span array types to store target document texts and their tokenizations.
+#
+
+from typing import *
+
+from text_extensions_for_pandas import SpanArray
+from text_extensions_for_pandas.array.thing_table import ThingTable
+
+
+class _BoxedSpanArray:
```

Comment:
Is the reason we need this because we are putting tokens as a `SpanArray` in here?

Reply:
Yup. The `__eq__` method on an ExtensionArray returns a mask. Python dictionaries don't like that.